ops/buildkite/deploy-content-preview

```diff
@@ -0,0 +1,163 @@
+#  useful for maintaining a stable review URL across time for a complex feature
+#  spread across multiple stacked PRs. For example, if you want to deploy to
+#  https://donate.owid.pages.dev, while on the `donate-2` branch you can run:
+#
+#  PREVIEW_BRANCH=donate yarn deploy-content-preview
+#
+#  Note that with or without the PREVIEW_BRANCH argument, this script will
+#  deploy from the staging server matching your current git branch name. For
+#  example, if you're on the `donate` branch, the script will SSH into the
+#  `staging-site-donate` staging server and execute the deployment commands from
+#  there.
+#
+#  # SETUP (only once)
+#
+#  ## RE-BAKING CONTENT WITH THE CORRECT BAKED_BASE_URL
+#
+#  This script doesn't bake content, it only deploys the already baked content
+#  in ~/live-date/bakedSite/ after performing some post-bake transformation
+#  steps, and storing the result in ~/dist.
+#
+#  Originally, the content is baked for the staging URL (e.g.
+#  http://staging-site-donate).
+#
+#  Before running this script, you need to rebake the content using the target
+#  preview BAKED_BASE_URL (e.g. https://donate.owid.pages.dev) to avoid CORS
+#  issues. This is done by running the following sequence on the staging server
+#  (not locally):
+#  - in ~/owid-grapher/.env, set BAKED_BASE_URL to
+#    https://[PREVIEW_BRANCH].owid.pages.dev
+#  - `rm ~/.site-baked` (see ops > templates > lxc-manager > grapher-refresh)
+#  - locate the latest Buildkite job for your staging environment and click
+#    "Rebuild". This will run the full site baking sequence with the preview URL
+#    as BAKED_BASE_URL.
+#
+#  You don't need to continuously rebake the site to test iterative changes to
+#  functions code in the preview environment. Functions are not baked, they are
+#  rather merely copied as-is to the `dist` folder to be executed on the server.
+#  When git pushing functions code, the regular staging job will automatically
+#  update them on the staging server. You then just need to run `yarn
+#  deploy-content-preview` to deploy the updated functions code to the preview
+#  URL.
+#
+#  ## CREATING CLOUDFLARE PAGES API TOKEN
+#
+#  This script requires a Cloudflare API token to deploy to the preview URL.
+#
+#  In particular, `wrangler` uses CLOUDFLARE_ACCOUNT_ID and
+#  CLOUDFLARE_API_TOKEN, read from owid-grapher/.env on the staging server.
+#
+#  - CLOUDFLARE_ACCOUNT_ID: You can find your account ID at
+#  https://dash.cloudflare.com/ > Our World in Data. The account ID is in the
+#  URL, e.g. https://dash.cloudflare.com/1234567890abcdef1234567890abcdef
+#
+# - CLOUDFLARE_API_TOKEN: You can create your token at
+#  https://dash.cloudflare.com/profile/api-tokens. Use the "Create Custom Token"
+#  template, then give it the "Account > Cloudflare Pages > Edit" permission.
+#  Use a TTL corresponding to the duration of your test (e.g. 2 weeks).
+#
+#  Save both these values in owid-grapher/.env on the staging server (not
+#  locally).
```

Comment:
@Marigold I couldn't deploy from a staging server to cloudflare (preview) using CLOUDLFARE_* env vars copied over from `staging-site-master`, and set up on your account. I'm not sure what they are used for at the moment. For now, I've added instructions to create a new token but it might just be simpler to update your token to grant "Cloudflare pages" > "Edit" permissions? Also, I'm not sure if we can view those shared tokens on the dashboard (yours is not there for me)? For instance, do you see the one I created on the owid account?

Reply:
I don't see yours in my dashboard. The account number in `staging-site-master` was pointing to my personal CF account. I changed it to OWID one. Could you try it out?

Reply:
Yes that went through!

My preference would be to use a different token from the production one though, just so we could quickly discard the staging one in case it gets compromised, without impacting production. It might also be helpful to add it to 1password with the list of permissions granted, since you only have access to it at the moment.

Reply:
Can you maybe give a brief recap of how these tokens are managed in the dev weekly next week?

Reply:
@danyx23 sure! On a side note, [creating tokens is now unnecessary](https://github.com/owid/owid-grapher/blob/deploy-cloudflare-previews/ops/buildkite/deploy-content-preview#L80-L81) since @Marigold upgraded the `staging-site-master` tokens, which this script sources.